### ISSUE TITLE
[FR] add option for gradient accumulation

### ISSUE THREAD

**Issue by PhilipMay** [UKPLab/sentence-transformers#1752]
I would like to have an option to do gradient accumulation.

My experiments show that large batch sizes are good. For example when you use `MultipleNegativesRankingLoss`.

My experiments also show that using amp increases the possible max. batch size but also decreases perforformance.

An other option would be multi GPU training but AFAIK @nreimers made experiments that show that multi GPU does not help...

So I would like to suggest an option for gradient accumulation.

**Comment by nreimers** [comment]
Gradient accumulation doesn't help for MultipleNegativesRankingLoss. The benefit from larger batches comes from more in-batch negatives. Gradiant acc. just computes two losses, averages them and then does back prop. The result will be the same.

**Comment by PhilipMay** [comment]
Yes. That sounds like it is right. So try multi GPU again @nreimers ?

Do you know why multiple GPUs did not work?

PS: Should I close this?

**Comment by nreimers** [comment]
For Multi GPU training, you can give this PR a try:
https://github.com/UKPLab/sentence-transformers/pull/1215

Sadly upgrading the losses to multi GPU can be quite complex and error prone.
